R/retrieve.R

```diff
@@ -0,0 +1,184 @@
+  dataset_info <- all_datasets[which(all_datasets$name == dataset), ]
+  if (nrow(dataset_info) == 1) {
+    group_path <- retrieve_value_key(dataset_info$group)
+    category_path <- retrieve_value_key(dataset_info$category)
+    dataset_path <- paste0(dataset, ext[[dataset_info$group]])
+    file_path <- file.path(
+      base_path, group_path,
+      category_path, dataset_path
+    )
+  } else {
+    file_path <- file.path(NULL)
+  }
+  if (rlang::is_empty(file_path)) {
+    stop("`dataset` not found")
+  }
+  return(file_path)
+}
+
+#' Retrieve climate directory path
+#'
+#' @description
+#' Climate data is retrieved from a general directory. Path is build for said
+#' directory
+#'
+#' @return character with path to retrieve the dataset from server
+#'
+#' @keywords internal
+retrieve_climate_path <- function() {
+  base_path <- retrieve_value_key("base_path")
+  group_path <- retrieve_value_key("climate")
+  directory_path <- file.path(base_path, group_path)
+  return(directory_path)
+}
+
+#' Retrieve dictionary path of named dataset
+#'
+#' @param dataset character with the dictionary name
+#'
+#' @description
+#' Dictionaries are not included in the general documentation file. Therefore,
+#' the path is built internally
+#'
+#' @return character with path to retrieve the dataset from server
+#'
+#' @keywords internal
+retrieve_dict_path <- function(dataset) {
+  base_path <- retrieve_value_key("base_path")
+  group_path <- retrieve_value_key("dictionaries")
+  dataset_path <- sprintf("%s.csv", dataset)
+  file_path <- file.path(base_path, group_path, dataset_path)
+  return(file_path)
+}
+
+#' Retrieve support dataset path
+#'
+#' @param dataset character with the support dataset name
+#'
+#' @description
+#' Support data is used for internal purposes and they are not included in the
+#' general documentation file
+#'
+#' @return character with path to retrieve the dataset from server
+#'
+#' @keywords internal
+retrieve_support_path <- function(dataset) {
+  base_path <- retrieve_value_key("base_path")
+  dataset_path <- retrieve_value_key(dataset)
+  file_path <- file.path(base_path, dataset_path)
+  if (rlang::is_empty(file_path)) {
+    stop("`dataset` not found")
+  }
+  return(file_path)
+}
+
+#' Retrieve table (csv and data) file
+#'
+#' @param dataset_path character path to the dataset on server
+#'
+#' @param sep separator for table data
+#'
+#' @return \code{data.frame} object with downloaded data
+#'
+#' @keywords internal
+retrieve_table <- function(dataset_path, sep = ";") {
+  request <- httr2::request(base_url = dataset_path)
+  response <- httr2::req_perform(request)
+  content <- httr2::resp_body_string(response)
+  downloaded_data <- suppressMessages(
+    suppressWarnings(
+      readr::read_delim(content,
+        delim = sep,
+        escape_double = FALSE,
+        trim_ws = TRUE,
+        show_col_types = FALSE
+      )
+    )
+  )
+  downloaded_data <- as.data.frame(downloaded_data)
+  return(downloaded_data)
+}
```

Comment:
I can't test it right now but most csv reading functions can read directly from URLs, without the need for httr2. Could you double check please?

Reply:
> I can't test it right now but most csv reading functions can read directly from URLs, without the need for httr2. Could you double check please?

Indeed, it can be changed. We used to have it because of how data was previously thought to be accessed; however, it can be removed now. This is addressed in #123 .